ui/src/components/addLiquidity.tsx

```diff
@@ -0,0 +1,236 @@
+  // Create useful methods on our mint address
+  const baseMintAddress = useCurrencyLeg(props.baseMintAddress);
+  const outcome0 = useCurrencyLeg(props.outcomes[0].mint_pk);
+  const outcome1 = useCurrencyLeg(props.outcomes[1].mint_pk);
+  const baseMint = useMint(props.baseMintAddress);
+  const outcome0Mint = useMint(props.outcomes[0].mint_pk);
+  const outcome1Mint = useMint(props.outcomes[1].mint_pk);
+  const [options, setOptions] = useState<PoolConfig>({
+    curveType: 0,
+    tradeFeeNumerator: 25,
+    tradeFeeDenominator: DEFAULT_DENOMINATOR,
+    ownerTradeFeeNumerator: 5,
+    ownerTradeFeeDenominator: DEFAULT_DENOMINATOR,
+    ownerWithdrawFeeNumerator: 0,
+    ownerWithdrawFeeDenominator: DEFAULT_DENOMINATOR,
+  });
+  const { pools } = useCachedPool();
+
+  const fundPool = async (components: Array<any>, pool: any) => {
+    pendingTxNum += 1;
+    // Add the liquidity to the first pool
+    addLiquidity(connection, wallet, components, slippage, pool, options)
+      .then(() => {
+        pendingTxNum -= 1;
+        if (pendingTxNum === 0) {
+          setPendingTx(false);
+        }
+      })
+      .catch((e) => {
+        console.log("Transaction failed", e);
+        notify({
+          description:
+            "Please try again and approve transactions from your wallet",
+          message: "Adding liquidity cancelled.",
+          type: "error",
+        });
+        pendingTxNum -= 1;
+        if (pendingTxNum === 0) {
+          setPendingTx(false);
+        }
+      });
+  }
+  const hasSufficientBalance = baseMintAddress.sufficientBalance()
+
+  function parseAmount(mint: MintInfo | undefined, amount: string) {
+    return parseFloat(amount) * Math.pow(10, mint?.decimals || 0);
+  }
+  // Swap usdc for market tokens
+  const executeAction =
+    !connected
+      ? wallet.connect :
+      async () => {
+        // TODO: Confirm mint address exists for outcome pk
+        // TODO: Should fail fast be implemented here i.e if one pool was not funded, can the other pool be ?
+        if (baseMintAddress.account && baseMintAddress.mint) {
+          setPendingTx(true);
+          pendingTxNum += 1;
+          // @ts-ignore
+          issueSet(props.market, parseAmount(baseMint, baseMintAddress.amount) / 2, wallet, connection)
+            .then(async () => {
+              // Check that the outcome token accounts exists
+              if (!outcome0.account || !outcome1.account) {
+                notify({
+                  description:
+                    "Token account does not exists",
+                  message: "Adding liquidity cancelled.",
+                  type: "error",
+                });
+              }
+              pendingTxNum -= 1;
+              // Fund pools of the outcome pk
+              [outcome0, outcome1].forEach(async (outcome, i) => {
+                PoolForBasketPromise([baseMintAddress.mintAddress, outcome.mintAddress].sort(), connection, pools).then(async (pool) => {
```

Comment:
why do you sort here? is it needed?

Reply:
Yes, it is. The usePoolForBasket() hook sorts the known pools when finding the pool for the corresponding token addresses. If I don't sort, it would not match any pool.